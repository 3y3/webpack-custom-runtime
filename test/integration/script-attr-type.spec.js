'use strict';

const path = require('path');
const diff = require('diff');
const webpack = require('./utils/webpack');
const WebpackCustomRuntime = require('../../').CustomRuntimePlugin;

const mainOutputDir = path.join(__dirname, 'fixtures', 'dist');

describe('ScriptAttr(Type)', function() {

    function cit(name, options, config) {
        test(name, async function() {
            const base = webpack({
                plugins: [
                    new WebpackCustomRuntime({ behavior: '' }),
                ]
            });
            const curr = webpack(Object.assign({
                plugins: [
                    new WebpackCustomRuntime({ behavior: '' }),
                    new WebpackCustomRuntime.ScriptAttr.Type(options),
                ]
            }, config));

            await Promise.all([ base.result, curr.result ]);

            const baseDir = base.fs.readdirSync(mainOutputDir);
            const currDir = curr.fs.readdirSync(mainOutputDir);

            expect(baseDir).toEqual(currDir);

            currDir.forEach((file) => {
                const linesDiff = diff.createTwoFilesPatch(
                    file, file,
                    base.fs.readFileSync(mainOutputDir + '/' + file).toString(),
                    curr.fs.readFileSync(mainOutputDir + '/' + file).toString(),
                    '', '',
                    { context: 0 }
                );

                expect(linesDiff).toMatchSnapshot(file);
            });
        });
    }

    cit('should not to override script type by default', {});

    cit('should override script type from options', { value: 'application/json' });

    cit('should override script type from config', {}, {
        output: {
            path: mainOutputDir,
            filename: '[name].js',
            jsonpScriptType: 'module'
        }
    });
});
