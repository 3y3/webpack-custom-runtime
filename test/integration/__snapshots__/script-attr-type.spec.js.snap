// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptAttr(Type) should override script type from config: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ Change 1 @@
-/******/ 	  return {};
+/******/ 	  mode = mode || 'load';
+/******/
+/******/ 	  var result = {};
+/******/
+/******/ 	  (function() {
+/******/ 	    result.type = 'module';
+/******/ 	  })();
+/******/
+/******/ 	  return result;
"
`;

exports[`ScriptAttr(Type) should override script type from options: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ Change 1 @@
-/******/ 	  return {};
+/******/ 	  mode = mode || 'load';
+/******/
+/******/ 	  var result = {};
+/******/
+/******/ 	  (function() {
+/******/ 	    result.type = 'application/json';
+/******/ 	  })();
+/******/
+/******/ 	  return result;
"
`;
